tests/SpecTests/Operation.php

```diff
@@ -337,16 +404,61 @@ private function executeForDatabase(Database $database, Context $context)
                     $args['pipeline'],
                     array_diff_key($args, ['pipeline' => 1])
                 );
+            case 'listCollections':
+                return $database->listCollections($args);
             case 'runCommand':
                 return $database->command(
                     $args['command'],
                     array_diff_key($args, ['command' => 1])
                 )->toArray()[0];
+            case 'watch':
+                return $database->watch(
+                    isset($args['pipeline']) ? $args['pipeline'] : [],
+                    array_diff_key($args, ['pipeline' => 1])
+                );
             default:
                 throw new LogicException('Unsupported database operation: ' . $this->name);
         }
     }
 
+    /**
+     * Executes the GridFS bucket operation and return its result.
+     *
+     * @param Bucket  $bucket
+     * @param Context $context Execution context
+     * @return mixed
+     * @throws LogicException if the database operation is unsupported
+     */
+    private function executeForGridFSBucket(Bucket $bucket, Context $context)
+    {
+        switch ($this->name) {
```

Comment:
> I suppose there's no reason to do this because we reference exact options in the cases below and aren't passing the full array to anything. I suppose it's something to keep in mind if/when GridFS gets session support, although who knows what that API may look like.

That is correct. We only use the `filename` option for the `download_by_name` operation and leave all other options unused. We could pass in options on the off-chance that a test specifies the `revision` option (which currently is the only option taken by these GridFS operations).